### PR TITLE
fragment iteration to support fully unrolled mma ops

### PR DESCRIPTION
**Pull request by shmsong**
This PR concerns iterating over registers of operands with mma ops. The difficulty is that the in-register layout of mma operands are different from the assumed root domain layout, especially when ldmatrix is used. The current `fragment iterator` is a workaround to get to a reasonable state to unblock forward progress, while avoiding over-complicating index math near term. 

Eventually much of the logic would be understood by iterdomain graph once the swizzle logic is fully piped through in follow ups. But these future clean ups would only affect styling. The PTX emitted from the generated CUDA code would be the same as it is today. 

Also in this PR is some minor extension to the matmul scheduler to configure double buffering of the intermediate tensors.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/index_compute.cpp [hunk 1]
```diff
@@ -416,6 +416,37 @@ Val* getProducerIndexWithPartialSplit(
       producer_index, SimplifyingIrBuilder::create<Int>(diff_eval.value()));
 }
 
+Val* getAllocSizeInMmaFragments(
```
Comment (shmsong) [inline torch/csrc/jit/codegen/cuda/index_compute.cpp]: Thinking about getting rid of this.

Comment (shmsong) [reply torch/csrc/jit/codegen/cuda/index_compute.cpp]: Alternative approach would just be to insert another set.

Comment (shmsong) [reply torch/csrc/jit/codegen/cuda/index_compute.cpp]: Removed.

**Review by csarofeen — APPROVED** [review]
LGTM

**Pull request merged**